PhysicsTools/PyTorch/plugins/PyTorchService.cc

```diff
@@ -0,0 +1,31 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+#include "FWCore/ServiceRegistry/interface/GlobalContext.h"
+#include "FWCore/ServiceRegistry/interface/ServiceMaker.h"
+#include "PhysicsTools/PyTorch/interface/TorchCompat.h"
+
+class PyTorchService {
+public:
+  PyTorchService(const edm::ParameterSet& config, edm::ActivityRegistry& registry) {
+    registry.watchPreGlobalBeginRun(this, &PyTorchService::preGlobalBeginRun);
```

Comment:
How about moving this even earlier in the initialisation, like
```suggestion
    registry.watchPreallocate(this, &PyTorchService::preallocate);
```
?

The syntax should be
```c++
void preallocate(edm::service::SystemBounds const&) {
  ...
}
```